src/custom/connectors/index.ts

```diff
@@ -1,16 +1,110 @@
+
+type RpcNetworks = { [chainId: number]: string }
+
+function getRpcNetworks(): [RpcNetworks, number[]] {
+  // Make sure the mandatory envs are present
+  if (typeof SUPPORTED_CHAIN_IDS === 'undefined') {
+    throw new Error(`REACT_APP_NETWORK_URL must be a defined environment variable`)
+  }
+
+  // Get list of supported chains
+  const chainIds = SUPPORTED_CHAIN_IDS.split(',').map(chainId => Number(chainId.trim()))
+  if (chainIds.length === 0) {
+    throw new Error(`At least one network should be supported. REACT_APP_CHAIN_ID`)
+  }
+
+  // Make sure the default chain is in the list of supported chains
+  if (!chainIds.includes(DEFAULT_CHAIN_ID)) {
+    throw new Error(
+      `The default chain id (${DEFAULT_CHAIN_ID}) must be part of the list of supported networks: ${chainIds.join(
+        ', '
+      )}`
+    )
+  }
+
+  // Return rpc urls per network
+  const rpcNetworks = chainIds.reduce<RpcNetworks>((acc, chainId) => {
+    const url = process.env['REACT_APP_NETWORK_URL_' + chainId]
+
+    if (typeof url === 'undefined') {
+      throw new Error(
+        `Network ${chainId} is supported, however 'REACT_APP_NETWORK_URL_${chainId} environment variable was not defined`
+      )
+    }
+
+    acc[chainId] = url
+
+    return acc
+  }, {})
+
+  // Get chainIds (excluding the DEFAULT_CHAIN_ID)
+  // Reason: By convention we will return DEFAULT_CHAIN_ID as the first element in the supported networks
+  const otherChainIds = Object.keys(rpcNetworks)
+    .map(Number)
+    .filter(networkId => networkId !== DEFAULT_CHAIN_ID)
+  const supportedChainIds = [DEFAULT_CHAIN_ID, ...otherChainIds]
+
+  return [rpcNetworks, supportedChainIds]
+}
+
+const [rpcNetworks, supportedChainIds] = getRpcNetworks()
+
+export const network = new NetworkConnector({
+  urls: rpcNetworks,
+  defaultChainId: NETWORK_CHAIN_ID
+})
+
+let networkLibrary: Web3Provider | undefined
+export function getNetworkLibrary(): Web3Provider {
+  return (networkLibrary = networkLibrary ?? new Web3Provider(network.provider as any))
+}
+
+export const injected = new InjectedConnector({ supportedChainIds })
+
+// mainnet only
+export const walletconnect = new WalletConnectConnector({
+  // TODO: Use any network when this PR is merged https://github.com/NoahZinsmeister/web3-react/pull/185
+  // rpc: rpcNetworks,
+  rpc: { 1: rpcNetworks[DEFAULT_CHAIN_ID] },
+  bridge: 'https://bridge.walletconnect.org',
+  qrcode: true,
+  pollingInterval: 15000
+})
+
+// mainnet only
+export const fortmatic = new FortmaticConnector({
+  apiKey: FORMATIC_KEY ?? '',
+  chainId: DEFAULT_CHAIN_ID
+})
+
+// mainnet only
+export const portis = new PortisConnector({
+  dAppId: PORTIS_ID ?? '',
+  // TODO: Allow to configure multiple networks in portis
+  // networks: supportedChainIds
+  networks: [DEFAULT_CHAIN_ID]
+})
+
+// mainnet only
+export const walletlink = new WalletLinkConnector({
+  url: rpcNetworks[DEFAULT_CHAIN_ID],
+  appName: 'Uniswap',
```

Comment:
appName here. was this changed in the other PR?

Reply:
yes, this was in another PR. keeping the scope just to the support of networks...

Reply:
anyways, i went for it for saving some merge conflicts...

Reply:
haha figured it'd be easier here.